R/driver-snowflake.R

```diff
@@ -87,3 +87,212 @@ setMethod("odbcDataType", "Snowflake",
     )
   }
 )
+
+#' Helper for Connecting to Snowflake via ODBC
+#'
+#' @description
+#'
+#' Connect to a Snowflake account via the [Snowflake ODBC
+#' driver](https://docs.snowflake.com/en/developer-guide/odbc/odbc).
+#'
+#' In particular, the custom `dbConnect()` method for the Snowflake ODBC driver
+#' detects ambient OAuth credentials on platforms like Snowpark Container
+#' Services or Posit Workbench.
+#'
+#' @inheritParams DBI::dbConnect
+#' @param account A Snowflake [account
+#'   identifier](https://docs.snowflake.com/en/user-guide/admin-account-identifier),
+#'   e.g. `"testorg-test_account"`.
+#' @param driver The name of the Snowflake ODBC driver, or `NULL` to use the
+#'   default name.
+#' @param warehouse The name of a Snowflake compute warehouse, or `NULL` to use
+#'   the default.
+#' @param database The name of a Snowflake database, or `NULL` to use the
+#'   default.
+#' @param schema The name of a Snowflake database schema, or `NULL` to use the
+#'   default.
+#' @param uid,pwd Manually specify a username and password for authentication.
+#'   Specifying these options will disable ambient credential discovery.
+#' @param ... Further arguments passed on to [`dbConnect()`].
+#'
+#' @returns An `OdbcConnection` object with an active connection to a Snowflake
+#'   account.
+#'
+#' @examples
+#' \dontrun{
+#' # Use ambient credentials.
+#' DBI::dbConnect(odbc::snowflake())
+#'
+#' # Use browser-based SSO (if configured). Only works on desktop.
+#' DBI::dbConnect(
+#'   odbc::snowflake(),
+#'   account = "testorg-test_account",
+#'   authenticator = "externalbrowser"
+#' )
+#'
+#' # Use a traditional username & password.
+#' DBI::dbConnect(
+#'   odbc::snowflake(),
+#'   account = "testorg-test_account",
+#'   uid = "me",
+#'   pwd = rstudioapi::askForPassword()
+#' )
+#' }
+#' @export
+snowflake <- function() {
+  new("Snowflake")
+}
+
+#' @rdname snowflake
+#' @export
+setMethod(
+  "dbConnect", "Snowflake",
+  function(drv,
+           account = Sys.getenv("SNOWFLAKE_ACCOUNT"),
+           driver = NULL,
+           warehouse = NULL,
+           database = NULL,
+           schema = NULL,
+           uid = NULL,
+           pwd = NULL,
+           ...) {
+    args <- snowflake_args(
```

Comment:
Need to add basic input validation here

Reply:
I don't think we do that in `odbc::databricks()` either but could add it; do you have suggestions?

Reply:
See [these lines](https://github.com/r-dbi/odbc/blob/2310dae3b12e53564e0baa495463a6dd41fddcd9/R/driver-databricks.R#L68-L76)! Those are recent changes.